app/controllers/api/base_controller.rb

```diff
@@ -37,6 +37,24 @@ def verify_mfa_requirement
     render plain: "Gem requires MFA enabled; You do not have MFA enabled yet.", status: :forbidden
   end
 
+  def response_with_warnings(response)
+    response_with_mfa_warning(response)
+  end
+
+  def response_with_mfa_warning(response)
+    message = response
+    if @api_key.user.mfa_recommended_not_yet_enabled?
+      message += "\n\n[WARNING] For protection of your account and gems, we encourage you to set up multifactor authentication " \
+                 "at https://rubygems.org/multifactor_auth/new. Your account will be required to have MFA enabled in the future."
+    elsif @api_key.user.mfa_recommended_weak_level_enabled?
+      message += "\n\n[WARNING] For protection of your account and gems, we encourage you to change your multifactor authentication " \
+                 "level to UI and gem signin or UI and API at https://rubygems.org/settings/edit. " \
```

Comment:
I think we should single quote the levels to distinguish them from the rest of the text. Hence
> level to UI and gem signin or UI and API

would be

> level to 'UI and gem signin' or 'UI and API'